tests/test_processes.py

```diff
@@ -322,10 +322,10 @@ def run(self, **kwargs):
     def test_kill(self):
         proc: Process = utils.DummyProcess()
 
-        msg = MessageBuilder.kill(text='Farewell!')
-        proc.kill(msg)
+        msg_text = 'Farewell!'
+        proc.kill(msg_text=msg_text)
         self.assertTrue(proc.killed())
-        self.assertEqual(proc.killed_msg(), msg)
+        self.assertEqual(proc.killed_msg()[MESSAGE_KEY], msg_text)
```

Comment:
I don't understand this change 🤔 MESSAGE_KEY is imported and passed as key, why is that?

Reply:
It is basically the same as previous test. The previous one build the `msg` which is a dict and I use it to assert. Now, the msg is not build, so I directly compare the `msg_text`. The different here is with new change, the `proc.kill` only require passing the `msg_text`, rather than construct the message type first.

Reply:
The point is is you put these keys outside the only class that basically defines them, them wherever you want to use it, you have to import. And it's not clear at all, that these keys are keys for the dictionary made by  `MessageBuilder` class.

```python
from plumpy.process_comms import MESSAGE_TEXT_KEY, FORCE_KILL_KEY, INTENT_KEY

msg = proc.killed_msg()
msg_text = msg[MESSAGE_TEXT_KEY]

kill_flag = msg[FORCE_KILL_KEY]
```

whilst you could:

```python
from plumpy.process_comms import MessageBuilder as mb

msg = proc.killed_msg()
msg_text = msg[mb.MESSAGE_TEXT_KEY]

kill_flag = msg[mb.FORCE_KILL_KEY]
```

Reply:
```python
msg = proc.killed_msg()
msg_text = msg['MESSAGE_TEXT']
```

I think this is not good since the typo in the str would be checked by the pre-commit or modern IDE. Meanwhile, we did struggle a lot by that the exception from plumpy is not well propagated to the aiida log which makes such error even hard to find and debug. Usually such exception can cause the test in aiida-core hang up rather than raising exceptions. 

```python
msg = proc.killed_msg()
msg_text = msg[mb.MESSAGE_TEXT_KEY]
```

`mb` can be quite confuse, it looks more like megabytes from first glimpse. 

The proper solution will be defining in `plumpy.process_comms` a standard message protocol and hasing message defined as a dataclass then used in plumpy and se/de -serialize the message through for example msgpack. It is a generic pattern, which can be found such as in redis or bit-torrent.
But change the message to dataclass require also change the de/se function which is beyond the scope of this PR.